component-catalog-connectors/artifactory-connector/artifactory_catalog_connector/artifactory-catalog.json

```
@@ -0,0 +1,170 @@
+  "title": "Artifactory Component Catalog",
+  "name": "artifactory-catalog",
+  "schemaspace": "component-catalogs",
+  "schemaspace_id": "8dc89ca3-4b90-41fd-adb9-9510ad346620",
+  "metadata_class_name": "elyra.pipeline.component_metadata.ComponentCatalogMetadata",
+  "uihints": {
+    "title": "Artifactory Component Catalog",
+    "icon": "",
+    "reference_url": "https://github.com/elyra-ai/examples/tree/master/component-catalog-connectors/artifactory-connector"
+  },
+  "properties": {
+    "schema_name": {
+      "title": "Schema Name",
+      "description": "The schema associated with this instance",
+      "type": "string",
+      "const": "artifactory-catalog"
+    },
+    "display_name": {
+      "title": "Display Name",
+      "description": "Display name of this Component Catalog",
+      "type": "string",
+      "minLength": 1
+    },
+    "metadata": {
+      "description": "Additional data specific to this metadata",
+      "type": "object",
+      "properties": {
+        "description": {
+          "title": "Description",
+          "description": "Description of this Component Catalog",
+          "type": "string",
+          "default": "Artifactory component catalog"
+        },
+        "runtime_type": {
+          "title": "Runtime",
+          "description": "List of runtime types this catalog supports",
+          "type": "string",
+          "enum": [
+            "KUBEFLOW_PIPELINES"
+          ],
+          "default": "KUBEFLOW_PIPELINES",
+          "uihints": {
+            "field_type": "dropdown"
+          }
+        },
+        "categories": {
+          "title": "Category Names",
+          "description": "Assign the components in the catalog to one or more categories, to group them in the visual pipeline editor palette.",
+          "type": "array",
+          "items": {
+            "type": "string",
+            "maxLength": 18
+          },
+          "uihints": {
+            "field_type": "array",
+            "category": "Component Categories"
+          }
+        },
+        "artifactory_url": {
+          "title": "Artifactory URL",
+          "description": "URL of the Artifactory server",
+          "type": "string",
+          "format": "uri",
+          "uihints": {
+            "category": "Source",
+            "placeholder": "https://example.org/artifactory/"
+          }
+        },
+        "artifactory_username": {
+          "title": "Artifactory Username",
+          "description": "Username for the Artifactory server",
+          "type": "string",
+          "uihints": {
+            "category": "Source"
+          }
+        },
+        "artifactory_password": {
+          "title": "Artifactory Password/API-Key",
+          "description": "Password or API Key for the Artifactory server",
+          "type": "string",
+          "uihints": {
+            "secure": true,
```

Version 3.9 includes UI changes, which use a few different property names/values:

```suggestion
            "ui:field": "password",
```

As is, the input is not masked.

---

@ptitzler should I include both `"secure": true` and `  `"ui:field": "password"` so that the passwords are hidden in both pre 3.8 and 3.9+?

---

I have included both in the latest push is this sensible?

---

Yep! Semantically it's a bit confusing (because both properties mean the same) but it provides backward compatibility. The other option would be to label the connector as "supports Elyra 3.9+" and only use the new property name/value in the metadata schema. Based on the small number of problem reports and enhancement requests related to custom components, I don't believe that there are many users today that would be 'locked out' if we did that.